account_asset_management/models/date_range.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2009-2017 Noviat
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import time
+
+from odoo import api, models
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+class DateRange(models.Model):
+    _inherit = 'date.range'
+
+    @api.model
+    def create(self, vals):
+        # TODO:
+        # change logic to avoid table recompute overhead
+        # when a regular (duration = 1 year) new FY is created
+        fy_types = self.env['date.range.type'].search(
+            [('fiscal_year', '=', True)])
+        if vals.get('type_id') in fy_types._ids:
+            recompute_vals = {
+                'reason': 'creation of fiscalyear %s' % vals.get('code'),
+                'company_id':
+                    vals.get('company_id') or
+                    self.env.user.company_id.id,
+                'date_trigger': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT),
+                'state': 'open',
+            }
+            self.env['account.asset.recompute.trigger'].sudo().create(
+                recompute_vals)
+        return super(DateRange, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        fy_types = self.env['date.range.type'].search(
```

Comment:
same, also it seems that most of the code can be reused

Reply:
I agree but ... this code works like this for severals years at a lot of customer sites (in 6.2, 7.0, 8.0, 10.0) hence I prefer that we first merge this PR so that we have a stable 11.0 version for the account_asset_management module and than work on  a PR on top of the merged code and review the whole recompute logic. Every module in the OCA can be improved but that should not block to already publish code that is working fine for a long time.

Reply:
definitely legit, but if we don't put a TODO for refactor/cleanup and we don't provide hints on what to change in all those places, likely no one is going to do it in the next future. And on v12 mig we'll be reviewing the same thing and will get the same answer.  :wink: